src/payment/strategies/humm/humm-payment-strategy.ts

```diff
@@ -0,0 +1,33 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { OrderRequestBody } from '../../../order';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { isOffsiteRedirectResponse } from '../../is-offsite-redirect-response';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { ExternalPaymentStrategy } from '../external';
+
+export default class HummPaymentStrategy extends ExternalPaymentStrategy {
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId}));
+        } catch (error) {
+            if (isOffsiteRedirectResponse(error)) {
+                if (!error.body.provider_data) {
+                    throw new PaymentArgumentInvalidError(['payment.provider_data']);
```

Comment:
Is there a scenario where we could get a redirect response without an `error.body.provider_data`?

It seems odd to raise a `PaymentArgumentInvalidError` given we're in control of the BE that should be providing that specific argument.

Reply:
no, we need it to continue, the values are required https://docs.shophumm.com.au/custom_integration/checkout_api.html